src/main/java/frc/robot/subsystems/gripper/GripperIOReal.java

```diff
@@ -1,18 +1,77 @@
 package frc.robot.subsystems.gripper;
 
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicExpoTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.units.Angle;
 import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import lib.Utils;
 
 public class GripperIOReal implements GripperIO {
+    private final TalonFX angleMotor;
+    private final CANSparkMax rollerMotor;
+    private final DigitalInput sensor;
+    private final DutyCycleEncoder absoluteEncoder = new DutyCycleEncoder(0);
```

Comment:
Add an input for the encoder position.

Reply:
3d71c0d9a9065263a0852481c2a35439f455bb6a